libraries/testAutomation/testAutomation/test_runner.py

```diff
@@ -212,6 +214,18 @@ def parse_args() -> argparse.Namespace:
     parser = argparse.ArgumentParser(
         description="Process inputs for test automation runner"
     )
+    parser.add_argument(
```

Comment:
I'm getting a code smell for a leaky abstraction here where tests define matrix but the test runner also chooses sources/targets.  See Joel on Software's post The Law of Leaky Abstractions [1] for more details.

I like the idea of the test runner discovering and deploying all configurations that are needed rather than having an external system that has to know the matrix of what it will run.  What are you thinking longer term for these tests?

- [1] https://www.joelonsoftware.com/2002/11/11/the-law-of-leaky-abstractions/

Reply:
@lewijacn Not a blocker, What do you think of adding a diagram of what parts of this system drive what environments to spin up and select what tests to run?

IMO this would really our understanding of the test orchestration that will get more important when we add in argo or other layers of abstraction in the deployment & execution flow in MA.

Reply:
I agree some sort of documentation/diagram could add some clarity here, my gut reaction is to wait until we have more clarity on the argo piece before investing too much in this structure right now as this piece could change a good amount